### PR TITLE
refactor: Replace usages of the deprecated `setProviderType`

### PR DESCRIPTION
**Pull request by Gudahtt**
## **Description**

All usages of the deprecated `NetworkController` method `setProviderType` have been replaced. The method `setActiveNetwork` is now used instead; it was updated in a recent release to accept types as well as IDs.

## **Related issues**

N/A

## **Manual testing steps**

No functional changes

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by Gudahtt** [comment]
Blocked by next release of network controller

**Review by legobeat — DISMISSED** [review]
lgtm!

**Review by legobeat — DISMISSED** [review]
LGTM! Let's also rerun failing CI checks before merging.

**Review by hjetpoluru — DISMISSED** [review]
LGTM

**Review by vthomas13 — DISMISSED** [review]

**Comment by Gudahtt** [comment]
Rebased to resolve conflicts again

**Review by HowardBraham — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Review by zone-live — APPROVED** [review]

**Pull request merged**
